### ISSUE TITLE
Formatting Examples With New Printer Module

### ISSUE THREAD

**Issue by ghost** [pygae/galgebra#40]
I am working through all the examples with the new python 3 printer module.  When I finish I will send them to you.  I see the `collect` problem you mentioned in several examples (I am using sympy 1.4).  Here is the custom `collect` I wrote to solve the problem -

```
def Collect(A, exp_lst):
    (coefs,bases) = linear_expand(A)
    C = S(0)
    for x in exp_lst:
        if x in bases:
            i = bases.index(x)
            C += coefs[i]*x
    return C
```
I placed it in metric.py just after `linear_expand.`  I have replaced the `collect ` call in ga.py but not in mv.py yet.  Are there examples where calling `collect ` is a problem in mv.py that I can use to test.  I am only using your galgebra for coding now.  Would the best thing for me to do is just put a link to your github site in my github site and tell everone that your is the supported version of galgebra.

**Comment by utensil** [comment]
Hi Alan, my work on merging your new printer is tracked at #39. I need to make it compatible with Python 2.7 and I encountered a few bugs while verifying against all examples. I also have a newer dop.ipynb modified from yours which tests both `ga_expr` and `print("title=", ga_expr).

I didn't noticed your previous emails until today, I'm glad that you like our little efforts on GAlgebra and I'm trying to merge the glagebra3 you sent line by line but not quite there yet, ga.py is easy to merge but mv.py requires some efforts since I need to tell whether a line of difference is a feature or a fixed bug from the older version of galgebra. The work is tracked at branch abrombo-dev of pygae/galgebra . 

Also I think the best way to collaborate is that you use your new account abrombo to fork pygae/galgebra and send PRs or simply point me to your branch and I'll do cherry-pick or take other actions to merge your modification and ensure them pass test.

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by utensil** [comment]
P.S. I've also invited you to our slack channel clifforddev.slack.com . An email is supposed to be sent to your new email address with instructions to get on board.

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by abrombo** [comment]
I have forked your distribution to my github account and clone it to my test bed computer (I have a lot of computers at least 5).  I will make changes to the version on my test bed and push the changes to the fork on my github account.  The procedures are starting to come back to me.

A side question.  Apart from speed what is better about Julia versus Python?

**Comment by abrombo** [comment]
Here is my fork of your version:

https://github.com/abrombo/galgebra

**Comment by utensil** [comment]
> A side question. Apart from speed what is better about Julia versus Python?

Besides being able to generate native machine code by utilizing LLVM and provides a high-level way to write and profile the program, there're 3 more advantages to me in writing Julia:

1. The typing system: in Julia we can do all the fancy things with types and generic that's very important when implementing a Math library;
2. The typing system(pun intended): in Julia we can use more Unicode characters as identifiers or operators and type them like in LaTeX, it turns the experience of reading Julia to somewhat similar to reading Math;
3. The macros and metaprogramming: in Julia we can manipulate programs like manipulating data and it also saves verbosity and improves readability.

Plus it plays well with Python and inherits a lot from the Python ecosystem and this practice fits in the philosophy of "premature optimization is the root of all evil".

P.S. I left you a series of private messages in Slack in reply to your emails. (You can find them if you click me and "Message" or find me under "Direct Messages")

**Comment by abrombo** [comment]
I have pushed my changes to my fork of galgebra.

**Comment by utensil** [comment]
Sorry I haven't get to check out the fork and I'm struggling with my code merging the new printer and found some weird cases and try to reproduce them at https://nbviewer.jupyter.org/github/utensil/julia-playground/blob/master/py/reproduce_new_printer_bug.ipynb .

**Comment by utensil** [comment]
I'll try restarting the merge with your fork next week.

**Comment by abrombo** [comment]
No rush.  I just want printing multivectors in regular (.py) code and notebook (.ipynb) to look as much like printing python 3 types as possible.  Also in notebook to allow easy annotation of output in output cells and output of multiple objects in a single output cell.

**Comment by utensil** [comment]
The merge is tracked by  #39 and the `collect` issue is fixed in #48
